include/sound/sof/dai.h

```diff
@@ -76,7 +85,13 @@ struct sof_ipc_dai_config {
 		struct sof_ipc_dai_alh_params alh;
 		struct sof_ipc_dai_esai_params esai;
 		struct sof_ipc_dai_sai_params sai;
-	};
+	} dai_data[0];
+
+	/*
+	 * multi-dai stream maps. The ext_id used needs to match subdai_index
+	 * in each _params structure above
+	 */
+	struct sof_ipc_stream_map stream_map;
```

Comment:
I'm confused. This struct sof_ipc_dai_config is a part of the ABI, right? Isn't this change changing the size of the struct in some unclear way? Before we had a union above - i.e. the size that it took was equal to that of the largest member. Now it's a 0-size array, so, that size is 0. And instead you're adding a new struct after that... And 0-size arrays in the middle of a struct can have weird consequences... I'm really confused.

Reply:
ack, this is completely broken indeed. i started a parallel thread to remove all variable arrays from IPC structs.